src/reactivity/unwrap.ts

```diff
@@ -0,0 +1,51 @@
+import { isRef } from './ref'
+import { proxy, isFunction, isObject, isArray } from '../utils'
+import { isReactive } from './reactive'
+
+export function unwrapRefProxy(value: any) {
+  if (isFunction(value)) {
+    return value
+  }
+
+  if (isRef(value)) {
+    return value
+  }
+
+  if (isArray(value)) {
+    return value
+  }
+
+  if (isReactive(value)) {
```

Comment:
if you don't mind answering, I would really be grateful if you take the time

seems to me, that we have a base-case if `value` is `reactive`, and we don't recursively unwrap

but on the test-cases it doesn't seem to be the case:
https://github.com/vuejs/composition-api/pull/361/files#diff-ffb3fccc5cc98e58adb1477f88e460b83b048c4b693b3fdb479a559f0c66cb88R484-R490

https://github.com/vuejs/composition-api/pull/361/files#diff-ffb3fccc5cc98e58adb1477f88e460b83b048c4b693b3fdb479a559f0c66cb88R550-R552

asking because i'm running into a reproduction and for a minute i was tricked by a premature understanding of this check

Reply:
`reactive` has some caveats on [Ref Unwrapping](https://v3.vuejs.org/guide/reactivity-fundamentals.html#ref-unwrapping)

but if a value is reactive, you can safely assume that it's already unwrapping.

This commit is pretty old, some things changed since them